models/repo_list_test.go

```diff
@@ -60,10 +55,66 @@ func TestSearchRepositoryByName(t *testing.T) {
 		Page:     1,
 		PageSize: 10,
 		Private:  true,
-		Searcher: &User{ID: 14},
 	})
 
-	assert.NotNil(t, repos)
 	assert.NoError(t, err)
 	assert.Equal(t, int64(3), count)
+	assert.Len(t, repos, 3)
+
+	testCases := []struct {
+		name  string
+		opts  *SearchRepoOptions
+		count int
+	}{
+		{name: "PublicRepositoriesByName",
+			opts:  &SearchRepoOptions{Keyword: "big_test_", PageSize: 10},
+			count: 4},
+		{name: "PublicAndPrivateRepositoriesByName",
+			opts:  &SearchRepoOptions{Keyword: "big_test_", Page: 1, PageSize: 10, Private: true},
+			count: 8},
+		{name: "PublicAndPrivateRepositoriesByNameWithPagesizeLimitFirstPage",
+			opts:  &SearchRepoOptions{Keyword: "big_test_", Page: 1, PageSize: 5, Private: true},
+			count: 8},
+		{name: "PublicAndPrivateRepositoriesByNameWithPagesizeLimitSecondPage",
+			opts:  &SearchRepoOptions{Keyword: "big_test_", Page: 2, PageSize: 5, Private: true},
+			count: 8},
+		{name: "PublicRepositoriesOfUser",
+			opts:  &SearchRepoOptions{Page: 1, PageSize: 10, OwnerID: 15},
+			count: 3}, // FIXME: Should return 2 (only directly owned repositories), now includes 1 public repository from owned organization
```

Comment:
Same comment from above. Transitive owning.

Reply:
Organization and other repositories related to user should be included only if `collaborate` option is supplied.

Reply:
You are right. Should be `2` here.